documentation/dev_running_from_source.markdown

```diff
@@ -4,6 +4,25 @@ title: "Puppet Server: Running From Source"
 canonical: "/puppetserver/latest/dev_running_from_source.html"
 ---
 
+So you'd like to run Puppet Server from source?
+-----
+
+Before we get started, let's highlight a few gotchas that might save you some time.
+The things that you are most likely to run into trouble with are pretty much the
+same things that were tricky about running the Webrick server from source:
+
+* Knowing where your confdir/vardir/codedir are
+* Knowing what your certname is
+* Making sure that your certname is resolvable
+
+In Puppet Server, confdir/vardir/codedir *always* come from your config.  They
```

Comment:
I don't think this is true, is it?  Those settings aren't required, we still have defaults in code, I think.  The larger point is good but I think this sentence is a little misleading.